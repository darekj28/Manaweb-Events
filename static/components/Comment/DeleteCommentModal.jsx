var React = require('react');

export default class DeleteCommentModal extends React.Component {
	handleCommentDelete() {
		var obj = {feed_name : "BALT", 
					comment_id : this.props.comment.comment_id,
					unique_id : this.props.comment.unique_id};
		$.ajax({
			type : 'POST',
			url  : '/deleteComment',
			data : JSON.stringify(obj, null, '\t'),
			contentType: 'application/json;charset=UTF-8'
		});
		this.props.handleCommentDelete(this.props.comment);
	}
	render() {
		return(
			<div className="modal fade" id="DeleteCommentModal" aria-hidden="true">
				<div className="modal-dialog" role="document">
					<div className="modal-content">
						<div className="modal-header">
							<button type="button" className="close" data-dismiss="modal" aria-label="close">
								<span aria-hidden="true">&times;</span>
							</button>
							<h4 className="modal-title">Are you sure you want to delete?</h4>
						</div>
						<div className="modal-body" id="DeleteCommentModalBody">
							<button id='dcm_yes' type="button" className="btn btn-default" data-dismiss="modal" 
									onClick={this.handleCommentDelete.bind(this)}>Yes</button>
							<button id="dcm_no" type="button" className="btn btn-default" data-dismiss="modal">No</button>
						</div>
					</div>
				</div>
			</div>
		);
	}
}