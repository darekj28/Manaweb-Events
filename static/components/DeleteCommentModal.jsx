var React = require('react');
// var $ = require('jquery');

export default class DeleteCommentModal extends React.Component {
	constructor(props) {
		super(props);
		this.handleCommentDelete = this.handleCommentDelete.bind(this);
	}
	handleCommentDelete() {
		var obj = {feed_name : "BALT", 
					comment_id : this.props.comment.comment_id,
					unique_id : this.props.comment.unique_id};
		$.ajax({
			type : 'POST',
			url  : '/deleteComment',
			data : JSON.stringify(obj, null, '\t'),
			contentType: 'application/json;charset=UTF-8'
		});
		this.props.refreshFeed(this.props.comment.comment_id);
	}
	render() {
		return(
			<div className="modal fade" id="DeleteCommentModal" aria-hidden="true">
				<div className="modal-dialog" role="document">
					<div className="modal-content">
						<div className="modal-header">
							<button type="button" className="close" data-dismiss="modal" aria-label="close">
								<span aria-hidden="true">&times;</span>
							</button>
							<h4 className="modal-title">Are you sure you want to delete?</h4>
						</div>
						<div className="modal-body" id="DeleteCommentModalBody">
							<button id='dcm_yes' type="button" className="btn btn-default" data-dismiss="modal" 
									onClick={this.handleCommentDelete}>Yes</button>
							<button id="dcm_no" type="button" className="btn btn-default" data-dismiss="modal">No</button>
						</div>
					</div>
				</div>
			</div>
		);
	}
}