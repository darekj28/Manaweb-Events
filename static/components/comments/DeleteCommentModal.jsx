class DeleteCommentModal extends React.Component {
	constructor(props) {
		super(props);
		this.handleCommentDelete = this.handleCommentDelete.bind(this);
	}
	handleCommentDelete() {
		var obj = {feed_name : "BALT", unique_id : this.props.comment.unique_id};
		$.ajax({
			type : 'POST',
			url  : '/deleteComment',
			data : JSON.stringify(obj, null, '\t'),
			contentType: 'application/json;charset=UTF-8'
		});
		this.props.refreshFeed();
	}
	render() {
		return(
			<div className="modal fade" id="DeleteCommentModal" aria-hidden="true">
				<div className="modal-dialog" role="document">
					<div className="modal-content">
						<div className="modal-header">
							<button type="button" className="close" data-dismiss="modal" aria-label="close">
								<span aria-hidden="true">&times;</span>
							</button>
							<h4 className="modal-title">Are you sure you want to delete?</h4>
						</div>
						<div className="modal-body" id="DeleteCommentModalBody">
							<button type="button" className="btn btn-default" data-dismiss="modal" 
									onClick={this.handleCommentDelete}>Yes</button>
							<button type="button" className="btn btn-default" data-dismiss="modal">No</button>
						</div>
					</div>
				</div>
			</div>
		);
	}
}