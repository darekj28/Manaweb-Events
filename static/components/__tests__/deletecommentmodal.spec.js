import DeleteCommentModal from '../DeleteCommentModal.jsx';

describe('DeleteCommentModal', () => {
	it('should call handleCommentDelete when yes is clicked', () => {
		window.$ = {ajax: jest.fn()}
		const refreshFeed = jest.fn();
		const comment = {unique_id : 1};
		const wrapper = mount(
			<DeleteCommentModal comment={comment} refreshFeed={refreshFeed}/>
			);
		const btn = wrapper.find('#dcm_yes');
		btn.simulate('click');
		expect(refreshFeed).toHaveBeenCalled();
	});
	it('renders correctly', () => {
		const refreshFeed = jest.fn();
		const comment = {unique_id : 1};
		const dcm = renderer.create(
			<DeleteCommentModal comment={comment} refreshFeed={refreshFeed}/>
			);
		expect(dcm).toMatchSnapshot();
	});
})