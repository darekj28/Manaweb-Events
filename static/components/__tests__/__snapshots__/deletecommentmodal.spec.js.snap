exports[`DeleteCommentModal renders correctly 1`] = `
<div
  aria-hidden="true"
  className="modal fade"
  id="DeleteCommentModal">
  <div
    className="modal-dialog"
    role="document">
    <div
      className="modal-content">
      <div
        className="modal-header">
        <button
          aria-label="close"
          className="close"
          data-dismiss="modal"
          type="button">
          <span
            aria-hidden="true">
            Ã—
          </span>
        </button>
        <h4
          className="modal-title">
          Are you sure you want to delete?
        </h4>
      </div>
      <div
        className="modal-body"
        id="DeleteCommentModalBody">
        <button
          className="btn btn-default"
          data-dismiss="modal"
          id="dcm_yes"
          onClick={[Function]}
          type="button">
          Yes
        </button>
        <button
          className="btn btn-default"
          data-dismiss="modal"
          id="dcm_no"
          type="button">
          No
        </button>
      </div>
    </div>
  </div>
</div>
`;
